pandas/io/parsers.py

```diff
@@ -1772,7 +1773,21 @@ def _handle_usecols(self, columns, usecols_key):
                 col_indices = []
                 for u in self.usecols:
                     if isinstance(u, string_types):
-                        col_indices.append(usecols_key.index(u))
+                        if usecols_key.index(u) not in col_indices:
+                            col_indices.append(usecols_key.index(u))
+                        else: # found duplicates in usecols_key.
+                            # get indices of all instances of u in usecols_key. GH11823
+                            u_index = Index(usecols_key).get_loc(u)
+                            # convert the slice or  mask array to index
```

Comment:
this is way to complicated. Pls show the inputs in the cases (e.g. duplicate and non-duplicate).